python/S3_VPC_ENDPOINT_ENABLED/S3_VPC_ENDPOINT_ENABLED_test.py

```diff
@@ -0,0 +1,335 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+import sys
+import unittest
+try:
+    from unittest.mock import MagicMock
+except ImportError:
+    from mock import MagicMock
+import botocore
+
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EC2::VPC'
+
+#############
+# Main Code #
+#############
+
+CONFIG_CLIENT_MOCK = MagicMock()
+STS_CLIENT_MOCK = MagicMock()
+EC2_CLIENT_MOCK = MagicMock()
+
+class Boto3Mock():
+    @staticmethod
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        if client_name == 'ec2':
+            return EC2_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('S3_VPC_ENDPOINT_ENABLED')
+
+class ComplianceTest(unittest.TestCase):
+     # Unit test for no VPC is present -- GHERKIN Scenario 1
+    def test_scenario_1(self):
+        EC2_CLIENT_MOCK.reset_mock()
+        describevpc_return = {"Vpcs":[]}
+        EC2_CLIENT_MOCK.describe_vpcs = MagicMock(return_value=describevpc_return)
+        resp_expected = []
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        resp_expected.append(build_expected_response('NOT_APPLICABLE', '123456789012', compliance_resource_type='AWS::EC2::VPC'))
+        assert_successful_evaluation(self, response, resp_expected)
+
+    #Unit test for VPC is present and no VPC endpoints are present for that VPC -- GHERKIN Scenario 2
+    def test_scenario_2(self):
+        EC2_CLIENT_MOCK.reset_mock()
+        vpc_id = "vpc-1234567"
+        describevpc_return = {"Vpcs":[{"VpcId":vpc_id}]}
+        EC2_CLIENT_MOCK.describe_vpcs = MagicMock(return_value=describevpc_return)
+        describevpcendpoints_return = {"VpcEndpoints":[]}
+        EC2_CLIENT_MOCK.describe_vpc_endpoints = MagicMock(return_value=describevpcendpoints_return)
+        resp_expected = []
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'vpc-1234567', compliance_resource_type='AWS::EC2::VPC', annotation='There are no Amazon S3 VPC endpoints present in '+ vpc_id+'.'))
+        assert_successful_evaluation(self, response, resp_expected)
+
+    #Unit test for VPC is present and there are no S3 endpoint present for that VPC -- GHERKIN Scenario 2
+    def test_scenario_2_1(self):
+        EC2_CLIENT_MOCK.reset_mock()
+        vpc_id = "vpc-1234567"
+        describevpc_return = {"Vpcs":[{"VpcId":vpc_id}]}
+        EC2_CLIENT_MOCK.describe_vpcs = MagicMock(return_value=describevpc_return)
+        describevpcendpoints_return = {
+            "VpcEndpoints": [
```

Comment:
Remove unnecessary information in this JSON. It will make it clearer.

Reply:
@jongogogo sorry I didn't understand this. can you please elaborate

Reply:
done